crates/red_knot_test/src/lib.rs

```diff
@@ -136,7 +137,25 @@ fn run_test(db: &mut db::Db, test: &parser::MarkdownTest) -> Result<(), Failures
                 })
                 .collect();
 
-            let type_diagnostics = check_types(db, test_file.file);
+            let type_diagnostics = match catch_unwind(|| check_types(db, test_file.file)) {
+                Ok(type_diagnostics) => type_diagnostics,
+                Err(info) => {
+                    eprintln!("panic in test");
+                    let mut by_line = matcher::FailuresByLine::default();
+                    by_line.push(
+                        OneIndexed::from_zero_indexed(0),
+                        info.info
+                            .to_string()
+                            .split('\n')
+                            .map(String::from)
+                            .collect(),
```

Comment:
This does not include the full backtrace of the panic, since that seemed to be more verbose than we need at this point.  If we want it, we just change `info.info` to `info` here.